main_test.go

```diff
@@ -143,6 +216,52 @@ func waitForKubeconfigSecret(clusterName, namespace string) error {
 	}
 }
 
+func waitForNodesReady(clusterName, namespace string, expected int) error {
+	data, err := getChildKubeconfig(clusterName, namespace)
+	if err != nil {
+		return err
+	}
+	kubeconf, err := ioutil.TempFile("", "kubeconfig")
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer os.Remove(kubeconf.Name())
+	if err := ioutil.WriteFile(kubeconf.Name(), data, os.FileMode(0644)); err != nil {
+		return errors.WithStack(err)
+	}
+
+	readyRe := regexp.MustCompile(`\s+Ready`)
+	timeout := time.NewTimer(5 * time.Minute)
```

Comment:
I'd still recommend using `wait.PollImmediate` but if you want to keep this, we should defer stopping the timer & ticker.

Reply:
Agreed! It is finally time!